### ISSUE TITLE
Export users package.json

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#100]
Tool to craft a package.json file from the users installed packages ($HOME/lib/node_modules) and their package versions, so they can be easily reinstalled. This could also done automatically, or also done for the root partition (although it would not be necesary if used the one from the installer).

**Comment by netlovers** [comment]
This basically sounds like `npm install --save`. So installing a package should call this by default.

**Comment by piranna** [comment]
The problem is that each user packages are installed in a similar way to
global packages, and as far as I know there's no posibility of having a
package.json file for globally installed packages... :-/

**Comment by lite20** [comment]
@piranna the auto-updated could do this?

**Comment by piranna** [comment]
> @piranna the auto-updated could do this?

No, they are independent, this is mostly to have a list of your installed programs, packages and libraries with their version and quickly have them available on other machine.

**Comment by lite20** [comment]
Ah okay. How would we go about creating this?

**Comment by piranna** [comment]
> Ah okay. How would we go about creating this?

Don't know :-) But at this moment I've found you can get a list of the globally installed modules and their versions with `npm ls --depth 0`, don't know if this could be just enough because it's smells like a hack trick and there's the problem of how to invoke `npm` later to install them...
